src/Templates/ODataT4CodeGenerator.cs

```diff
@@ -1802,7 +1809,13 @@ internal void WriteNamespace(string fullNamespace)
 
                     if(context.GenerateMultipleFiles)
                     {
-                        context.MultipleFilesManager.StartNewFile($"{complexType.Name}{(this.context.TargetLanguage == LanguageOption.VB ? ".vb" : ".cs")}",false);
+                        string fileName = complexType.Name;
+                        if (schemaElementsInModel.Count(e => e.Name.Equals(complexType.Name)) > 1)
+                        {
+                            fileName = complexType.FullName();
+                        }
+
+                        context.MultipleFilesManager.StartNewFile($"{fileName}{(this.context.TargetLanguage == LanguageOption.VB ? ".vb" : ".cs")}", false);
```

Comment:
```suggestion
                        var fileExtension = this.context.TargetLanguage == LanguageOption.VB ? ".vb" : ".cs";
                        context.MultipleFilesManager.StartNewFile($"{fileName}{fileExtension}", false);
```
@unchase For readability

Reply:
I think should put it into the same helper method.